### PR TITLE
Implemented working sensor endpoint

### PR DESCRIPTION
**Pull request by Tais993**

### CODE REVIEW

File: Controllers/SensorController.cs [hunk 1]
```diff
@@ -9,40 +13,41 @@ public class SensorController
...
-			new Sensor
+		var sensors = queryAssets.Select(asset =>
+		{
+
+			// Extracting the coordinates array
+
+			var coordinatesInfo = ((JObject)asset.Attributes["location"].Value).ToObject<CoordinatesInfo>();
+
+
+			//TODO Fix vallue
```
Comment (Tais993) [inline Controllers/SensorController.cs]:
```suggestion
			//TODO Fix value
```

File: OpenRemoteAPI.csproj [hunk 2]
```diff
@@ -6,8 +6,11 @@
...
 
-    <ItemGroup>
-        <PackageReference Include="Swashbuckle.AspNetCore" Version="6.4.0"/>
-    </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
+    <PackageReference Include="Swashbuckle.AspNetCore" Version="6.4.0" />
+    <PackageReference Include="Swashbuckle.AspNetCore" Version="6.4.0"/>
+  </ItemGroup>
```
Comment (Tais993) [inline OpenRemoteAPI.csproj]:
```suggestion
```

File: OpenRemoteAPI.csproj [hunk 3]
```diff
@@ -6,8 +6,11 @@
         <ImplicitUsings>enable</ImplicitUsings>
     </PropertyGroup>
 
-    <ItemGroup>
-        <PackageReference Include="Swashbuckle.AspNetCore" Version="6.4.0"/>
-    </ItemGroup>
+
```
Comment (Tais993) [inline OpenRemoteAPI.csproj]:
```suggestion
```

File: Controllers/SensorController.cs [hunk 4]
```diff
@@ -9,40 +13,41 @@ public class SensorController
...
+			return new Sensor
 			{
-				Id = 1,
-				Name = "Example Sensor 1",
+				Id = asset.Id,
+				Name = asset.Name,
 				RoomId = 1,
-				Value = (float)Math.Round(rand.NextSingle(), 2),
+				Value = 0.5f,
```
Comment (thijnmens) [inline Controllers/SensorController.cs]: Sensor value still has static value

File: Internal/OpenRemoteAPI.cs [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        _httpClient.BaseAddress = new Uri(_config.BaseUrl);
+    }
+
+    internal static Config LoadConfig()
+    {
+        using StreamReader r = new StreamReader("./Properties/config.json");
+        string json = r.ReadToEnd();
+        return JsonConvert.DeserializeObject<Config>(json) ?? throw new InvalidOperationException("Missing config");
+    }
+
```
Comment (Tais993) [inline Internal/OpenRemoteAPI.cs:33]:
```suggestion


```

File: Internal/Models/Attribute.cs
Comment (thijnmens) [inline Internal/Models/Attribute.cs:1]: Missing getter setter, they were added in the other models so for consistency sake

File: Internal/OpenRemoteAPI.cs [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        string json = JsonConvert.SerializeObject(query, settings);
+        HttpContent httpContent = new StringContent(json, MediaTypeHeaderValue.Parse("application/json"));
+
+
+        HttpResponseMessage response = await MakeHttpCall(postAssetQuery.ToUrl(), postAssetQuery.HttpMethod, httpContent);
+
+        string readAsStringAsync = await response.Content.ReadAsStringAsync();
+        Console.WriteLine("Return JSON: " + readAsStringAsync);
+        List<Asset> assets = JsonConvert.DeserializeObject<List<Asset>>(readAsStringAsync) ?? [];
+        return assets;
```
Comment (Tais993) [inline Internal/OpenRemoteAPI.cs]:
```suggestion
        string json = JsonConvert.SerializeObject(query, settings);
        HttpContent httpContent = new StringContent(json, MediaTypeHeaderValue.Parse("application/json"));

        HttpResponseMessage response = await MakeHttpCall(postAssetQuery.ToUrl(), postAssetQuery.HttpMethod, httpContent);

        string readAsStringAsync = await response.Content.ReadAsStringAsync();
        return JsonConvert.DeserializeObject<List<Asset>>(readAsStringAsync) ?? [];
```

File: Internal/Models/CoordinatesInfo.cs
Comment (thijnmens) [inline Internal/Models/CoordinatesInfo.cs:1]: Either remove Coordinates.cs or remove this, we can't have 2 coordinate models

File: Internal/OpenRemoteAPI.cs [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+    public async Task<Asset?> QueryAsset(string assetId)
+    {
+        HttpResponseMessage response = await MakeHttpCall(getAsset.ToUrl(assetId), getAsset.HttpMethod);
+
+        string readAsStringAsync = await response.Content.ReadAsStringAsync();
+        Console.WriteLine("Return JSON: " + readAsStringAsync);
+        Asset? asset = JsonConvert.DeserializeObject<Asset>(readAsStringAsync);
+        return asset;
+    }
```
Comment (Tais993) [inline Internal/OpenRemoteAPI.cs]:
```suggestion
    public async Task<Asset?> QueryAsset(string assetId)
    {
        HttpResponseMessage response = await MakeHttpCall(getAsset.ToUrl(assetId), getAsset.HttpMethod);

        string readAsStringAsync = await response.Content.ReadAsStringAsync();
        return JsonConvert.DeserializeObject<Asset>(readAsStringAsync);
    }
```

File: Internal/OpenRemoteAPI.cs [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+    public async Task<HttpResponseMessage> MakeHttpCall(string routeUrl, HttpMethod httpMethod)
+    {
+        return await MakeHttpCall(routeUrl, httpMethod, null);
+    }
+
+    internal async Task<HttpResponseMessage> MakeHttpCall(string routeUrl, HttpMethod httpMethod, HttpContent httpContent)
+    {
+        Console.WriteLine("Calling `" + routeUrl + "` with " + httpMethod);
```
Comment (Tais993) [inline Internal/OpenRemoteAPI.cs]:
```suggestion
```

File: Internal/Requests/Route.cs [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+    {
+        public static Route getAssetDiscovery = new Route(GET, "agent/assetDiscovery/{agentId}");
+        public static Route assetImport = new Route(POST, "agent/assetImport/{agentId}");
+        public static Route getInstanceDiscovery = new Route(GET, "agent/instanceDiscovery/{agentId}");
+    }
+
+    public static class Configuration
+    {
+        public static Route postConfigurationFile = new Route(POST, "configuration/manager/file");
+        //TODO
```
Comment (thijnmens) [inline Internal/Requests/Route.cs]: Todo what?

**Review by thijnmens — CHANGES_REQUESTED** [review]

File: Controllers/SensorController.cs (hunk 4 above)
Comment (Tais993) [reply Controllers/SensorController.cs]: That will be fixed by @SlipperyTrout with another PR, so for now it's a static value.

File: Internal/Requests/Route.cs (hunk 9 above)
Comment (Tais993) [reply Internal/Requests/Route.cs]:
```suggestion
```

File: Controllers/SensorController.cs (hunk 4 above)
Comment (Tais993) [reply Controllers/SensorController.cs]: Will make it a random number for the time being

**Review by thijnmens — APPROVED** [review]

**Pull request merged**